### PR TITLE
fix(components): remove `animation` prop from all components

### PR DESCRIPTION
**Pull request by joschect**
# Breaking Changes
1. Animation property is being removed. Any component that makes use of the animation prop will break at build time. Any such components can be fixed by wrapping them in the existing `<Animation/>` component with the same animation props passed in.

### Before
```JSX
<Icon
    name="emoji"
    animation={{ name: 'spinner', delay: '5s', duration: '2s' }}
    circular
/>
```
### After
```JSX
<Animation name="spinner" delay="5s" duration="2s">
    <Icon name="emoji" circular />
</Animation>
```

## Why?
The animation prop added uneeded bulk to the api surface and would only be used in certain situations. Removing it will improve perf slightly, as it was adding another check to every single component render. Additionally a quick search has shown that this prop is not currently used and that the animation component is already favored.

### CODE REVIEW

**Review by khmakoto — APPROVED** [review]

**Review by aneeshack4 — APPROVED** [review]

File: packages/react/test/specs/utils/__snapshots__/felaRenderer-test.tsx.snap
Comment (layershifter) [inline packages/react/test/specs/utils/__snapshots__/felaRenderer-test.tsx.snap]: This looks broken, I expect that generated keyframes should stay there

**Review by layershifter — CHANGES_REQUESTED** [review]
Please check the issue with snapshots before merging

File: packages/react/test/specs/utils/__snapshots__/felaRenderer-test.tsx.snap
Comment (joschect) [reply packages/react/test/specs/utils/__snapshots__/felaRenderer-test.tsx.snap]: Thanks for the catch! I'll take a look

Comment (joschect) [reply packages/react/test/specs/utils/__snapshots__/felaRenderer-test.tsx.snap]: I've spent a good bit of time trying to figure this out, as far as I can tell it works, at least it does in all the examples that I've changed it is. But there is a breakdown in the fella-react-jest framework which is causing classnames to not get passed down. I'm wondering if React.Clone element isn't working/getting called in the Animation component. I'll take another look tomorrow.

Comment (miroslavstastny) [reply packages/react/test/specs/utils/__snapshots__/felaRenderer-test.tsx.snap]: `Animation` component depends on theme styles which I think is a bug.
Created an issue #2247 and pushed workaround for the snapshot test to this PR.

**Review by layershifter — APPROVED** [review]

**Review by mnajdova — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

**Comment by layershifter** [comment]
Is there something that blocks merging?

**Comment by joschect** [comment]
@layershifter Given the bug in Animation, I figured I should investigate that first. I've been looking into it. I guess I don't need a fix atm but I think it might be good to have a better idea as to why. That does seem like it might be an unrelated pr though.

**Comment by layershifter** [comment]
@joschect We had an offline conversation and agreed that we will fix linked issue separately, so there are no blockers for this PR 🎉

**Pull request merged**
